var/spack/repos/builtin/packages/genesis/package.py

```diff
@@ -49,44 +85,47 @@ class Genesis(AutotoolsPackage, CudaPackage):
 
     variant("openmp", default=True, description="Enable OpenMP.")
     variant("single", default=False, description="Enable single precision.")
+    variant("mixed", default=False, description="Enable mixed precision.", when="@2.0.0:")
     variant("hmdisk", default=False, description="Enable huge molecule on hard disk.")
 
     conflicts("%apple-clang", when="+openmp")
 
-    depends_on("autoconf", type="build", when="@1.5.1 %fj")
-    depends_on("autoconf", type="build", when="@master")
-    depends_on("automake", type="build", when="@1.5.1 %fj")
-    depends_on("automake", type="build", when="@master")
-    depends_on("libtool", type="build", when="@1.5.1 %fj")
-    depends_on("libtool", type="build", when="@master")
-    depends_on("m4", type="build", when="@1.5.1 %fj")
-    depends_on("m4", type="build", when="@master")
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+    depends_on("libtool", type="build")
+    depends_on("m4", type="build")
 
     depends_on("mpi", type=("build", "run"))
     depends_on("lapack")
 
+    depends_on("python@2.6.9:2.8.0", type=("build", "run"), when="@:1.7.1")
```

Comment:
Apologies for the delay, but by now no more python@2 dependencies are possible anymore for the current set of package versions in spack.